default.yml

```diff
@@ -0,0 +1,108 @@
+parameters:
+  minrust: 1.32.0 # Rust 2018 with uniform paths
+  setup: []
+  services: {}
+  env: {}
+  cross: true
+
+jobs:
+ - job: style
+   displayName: Style linting
+   strategy:
+     matrix:
+       stable:
+         rust: stable
+       beta:
+         rust: beta
+   pool:
+     vmImage: ubuntu-16.04
+   continueOnError: $[eq(variables.rust, 'beta')]
+   steps:
+     - template: install-rust.yml
+       parameters:
+         rust: $(rust)
+         components:
+           - rustfmt
+           - clippy
+     - script: cargo fmt --all -- --check
+       displayName: cargo fmt --check
+     - script: cargo clippy --all
+       displayName: cargo clippy
+ - job: main
+   displayName: Compile and test
+   dependsOn: []
+   ${{ if eq('true', parameters.cross) }}:
+     strategy:
+       matrix:
+         "Linux (nightly)":
+           vmImage: ubuntu-16.04
+           rust: nightly
+         "Linux (beta)":
+           vmImage: ubuntu-16.04
+           rust: beta
+         Linux:
+           vmImage: ubuntu-16.04
+           rust: stable
+         MacOS:
+           vmImage: macOS-10.14
+           rust: stable
+         Windows:
+           vmImage: windows-2019
+           rust: stable
+   ${{ if ne('true', parameters.cross) }}:
+     strategy:
+       matrix:
+         "Linux (nightly)":
+           vmImage: ubuntu-16.04
+           rust: nightly
+         "Linux (beta)":
+           vmImage: ubuntu-16.04
+           rust: beta
+         Linux:
+           vmImage: ubuntu-16.04
+           rust: stable
+   pool:
+     vmImage: $(vmImage)
+   services:
+     ${{ insert }}: ${{ parameters.services }}
+   continueOnError: $[eq(variables.rust, 'nightly')]
+   steps:
+     - template: install-rust.yml
+       parameters:
+         rust: $(rust)
+     # Run any user-specific setup steps
+     - ${{ parameters.setup }}
+     - script: cargo check --all --bins --examples
+       displayName: cargo check
+     - script: cargo check --all --bins --examples --no-default-features
+       displayName: cargo check --no-default-features
+     - script: cargo check --all --bins --examples --all-features
+       displayName: cargo check --all-features
+     - script: cargo test --all
+       displayName: cargo test
+       env:
+         ${{ insert }}: ${{ parameters.env }}
```

Comment:
If we're not testing under all feature combinations, we should probably include `--tests` under `cargo check`.

Reply:
Ah, that was a mistake, I intended for the default to be testing with all features.